fuel-poa-coordinator/Cargo.toml

```diff
@@ -0,0 +1,27 @@
+[package]
+name = "fuel-poa-coordinator"
+version = "0.10.1"
+authors = ["Fuel Labs <contact@fuel.sh>"]
+edition = "2021"
+homepage = "https://fuel.network/"
+keywords = ["blockchain", "fuel", "consensus", "bft"]
+license = "BUSL-1.1"
+repository = "https://github.com/FuelLabs/fuel-core"
+description = "Fuel Core PoA Coordinator"
+
+[dependencies]
+anyhow = "1.0"
+fuel-core-interfaces = { path = "../fuel-core-interfaces", version = "0.10.1" }
+humantime-serde = "1.1.1"
+parking_lot = "0.12"
+serde = { version = "1.0", features = ["derive"] }
+tokio = { version = "1.21", features = ["full"] }
+tracing = "0.1"
+
+[dev-dependencies]
+async-trait = "0.1"
+fuel-block-producer = { path = "../fuel-block-producer" }
+fuel-core = { path = "../fuel-core" }
```

Comment:
We should avoid depending on any other workspace members in these domain crates except for maybe `fuel-core-interfaces`.

Integration tests that fully depend on big dependencies like this should only exist in `fuel-tests`:

- This gives us assurances that the component actually works inside a real instance of fuel-core.
- The harness in fuel-tests is optimized to reduce the number of targets big dependencies like rocksdb must be linked into
- Changes to other crates like block producer or fuel-core won't drastically impact the structure of the domain-focused tests.


From what I can tell, the only reason we need fuel-core is to provide an impl of InterpreterStorage which has a method for returning the block height. However, the POA module shouldn't need to care about any of the other trait requirements of interpreter storage, as it never interacts with an instance of the VM.

The PoA orchestrator should have its own database trait for fetching the block height, and use a mock impl for testing instead of depending on fuel-core's implementation.

Reply:
Removed fuel-core dev-dependency and factored block height query to a separate trait.  The block producer dependency is removed as well, by simply copying the imported function as-is, which is actually nice as the test of poa-coordinator shouldn't depend on internals of the block producer.